ais/src/main/java/au/gov/amsa/ais/Util.java

```diff
@@ -274,6 +274,30 @@ public static void checkMessageId(int messageId, AisMessageType... messageTypes)
 			checkArgument(found, "messageId must be in [" + s + "]  but was " + messageId);
 		}
 	}
+	
+	/**
+	 * A check that the part number provided matches the collection of 
+	 * available AisStaticDataReportPart objects.
+	 * 
+	 * @param partNumber The part number.
+	 * @param partNumbers The collection of AisStaticDataReportPart objects.
+	 */
+	public static void checkStaticDataReportPartNumber(int partNumber, AisStaticDataReportPart... partNumbers) {
+		boolean found = false;
+		for (AisStaticDataReportPart reportPartNumber : partNumbers) {
+			if (reportPartNumber.getPartNumber() == partNumber)
+				found = true;
+		}
+		if (!found) {
+			StringBuffer s = new StringBuffer();
+			for (AisStaticDataReportPart reportPartNumber : partNumbers) {
+				if (s.length() > 0)
+					s.append(",");
+				s.append(reportPartNumber.getPartNumber() + "");
+			}
+			checkArgument(found, "part number must be in [" + s + "]  but was " + partNumber);
+		}
+	}
```

Comment:
Isn't this a bit wordy for what is really a check that the part number is one of two numbers?

Reply:
Fixed